### PR TITLE
Removed JQuery references - glotdict-settings.js

### LINKED ISSUE #339: Improve code of settings.js by replacing last jQuery occurrences
Replace the jQuery code present in the settings.js file with native JS while respecting the ESLint rules as well as the gd_ prefix for any global, variable or function.

### PR DESCRIPTION
**Pull request by Adnan-Sait**
Fixes #339

### CODE REVIEW

File: js/glotdict-settings.js [hunk 1]
```diff
@@ -18,21 +18,28 @@ if ( ( 'undefined' !== typeof $gp_editor_options ) && '1' === $gp_editor_options
...
+	const checkbox_listeners_flag = document.querySelector('.gp-content .gd_settings input[type="checkbox"]' ) ? true : false;
 	if ( document.body.classList.contains( 'gd-settings-on-screen' ) && '0' !== gd_extension.previousVersion ) {
 		gd_extension.previousVersion = gd_extension.currentVersion;
 		localStorage.setItem( 'gd_extension_status', JSON.stringify( gd_extension ) );
 	}
 	document.body.classList.toggle( 'gd-settings-on-screen' );
 	gd_generate_settings_panel();
-} );
 
-jQuery( '.gp-content' ).on( 'click', '.gd_settings input[type="checkbox"]', function() {
```
Comment (Mte90) [inline js/glotdict-settings.js:34]: I don't see a click event on this just a check of the checkbox status but is wrong we want that the value is changed on click not on the file loading.

**Review by Mte90 — CHANGES_REQUESTED** [review]

File: js/glotdict-settings.js (hunk 1 above)
Comment (Adnan-Sait) [reply js/glotdict-settings.js:34]: In both the current code and the PR, the click event handler is configured on the checkbox `.gp-content .gd_settings input[type="checkbox"]`. And when the checkbox is clicked we save the details and the state into the Local Storage (seen in line 35).

With Vanilla JS, the click event can only be configured after the checkboxes are rendered. This is the reason I have moved the checkbox code inside the `gd_settings_menu click function`. I have tested the checkbox functionality in different scenarios (opening and closing the GlotDict Settings), the code is working as required in all scenarios.

**Review by vlad-timotei — COMMENTED** [review]
Hi! 
Thank you for the PR.
I just tested your code and it works as expected. The events are added properly.

I have one question though.

What do you check with `checkbox_listeners_flag` flag? 
I see you check if there is at least one checkbox and if not only then add the event listeners.
Is there a reason for that? 

I think I see why, but please confirm. 😄 (To prevent multiple events being registered when a user opens and closes and reopens the Settings modal? 

Also, could you follow the eslint rules in the repo, just so it looks nice and consistent? :) 
The linter should fix it if you open it in a IDE.
- space after ( and before )
- single quote strings
- space after !
- arrow functions in callback

Thank you!

**Comment by Adnan-Sait** [comment]
@vlad-timotei , Yes I have added the `checkbox_listeners_flag` to ensure that the events are registered on the checkboxes only once (when the settings page is first opened).

I used ESLint in my IDE, it has updated the formatting in the file. I've pushed the changes to the branch.

Thank you 🙂

File: js/glotdict-settings.js (hunk 1 above)
Comment (Mte90) [reply js/glotdict-settings.js:34]: Yes you are right I missed that line.
we need some testings :-D

**Comment by vlad-timotei** [comment]
> Yes I have added the checkbox_listeners_flag to ensure that the events are registered on the checkboxes only once (when the settings page is first opened).

I know it's not in the scope of the initial issue, but what would you think about moving this part
```
checkbox.addEventListener( 'click', ( event ) => {
				localStorage.setItem( event.target.id, event.target.checked );
			} )
```
in `gd_generate_settings_panel()`?  This function has a safeguard check that will only allow it to run once, so we should be safe. This would be much more efficient to add the events when we create the element.

Somewhere at line 159

```
const input = document.createElement( 'INPUT' );
...
input.type = 'checkbox';
input.id = `gd_${setting_slug}`;
input.addEventListener(...)
```

This way we spare a few later queries.
What do you think?

**Review by vlad-timotei — APPROVED** [review]
Looks good to me now!
Thank you for your efforts!
Give us a few days just to run some tests on translate.w.org and I think it's good to go!

**Comment by Mte90** [comment]
I tested and everything works :-)

**Pull request merged**